### ISSUE TITLE
Using `delete` on a non-configurable property isn't legal

### ISSUE THREAD

**Issue by sudokode** [douglascrockford/JSLint#181]
Regarding the `Expected 'delete' and instead saw 'undefined'.` warning, this may not always be possible depending on the property settings. Is it possible to turn off just these sorts of `expected_a_b` warnings?

```javascript
(function () {
    'use strict';

    function Foo() {
        let self = Object.create(null);

        Object.defineProperty(self, 'foo', {
            configurable: false,
            writable: true
        });

        return self;
    }

    let test = new Foo();

    test.foo = 'bar';       // works
    test.foo = undefined;   // works
//             ^ Expected 'delete' and instead saw 'undefined'.

    delete test.foo;
//         ^ TypeError: property test.foo is non-configurable and can't be deleted
}());
```
Or is it better to just use `null` in these cases? I could imagine a scenario where a property's being `null` and its being `undefined` would make a difference.

**Comment by douglascrockford** [comment]
Storing 'undefined' in an object is at least confusing. Don't do that. And don't use inheritance badly. Write good programs and don't make bugs.
